pandas/core/indexes/numeric.py

```diff
@@ -220,7 +237,10 @@ def astype(self, dtype, copy=True):
                 # TODO(jreback); this can change once we have an EA Index type
                 # GH 13149
                 arr = astype_nansafe(self._values, dtype=dtype)
-                return Int64Index(arr, name=self.name)
+                if isinstance(self, Float64Index):
```

Comment:
why at this level?

Reply:
If we do `Float64Index().astype(int32)` we should get back an Int64Index[int64]. If we do `NumericIndex().astype(int32)` we should get a `NumericIndex[int32]`.

A fairly big bit of this type of complexity in this PR comes from dealing with NumericIndex vs. Int64Index etc.

Reply:
i don't understand your example if we now can astype to int32 why wouldn't we do that?

Reply:
why are Float64 and the existing Index types not just an alias for the NumericIndex of ty at type? 

eg it's just a name

Reply:
Currently both `Index().astype(int32)`, `Index(series, dtype=int32)` and `Float64().astype(int32)` return a `Int64Index`.

There may be quite some usages that expect this behavious to continue, so returning a `NumericIndex[int32]` would break in those cases.

Reply:
ok i c, can you leave some comments to this effect, also reference the issue about _is_backwards_compat_public_numeric_index (e.g. anywhere it should be eventually fixed)